web/db.py

```diff
@@ -1198,10 +1198,18 @@ def _process_insert_query(self, query, tablename, seqname):
                 seqname = None
 
         if seqname:
-            query += "; SELECT currval('%s')" % seqname
+            query += self.get_sequence_query(seqname)
 
         return query
 
+    def get_sequence_query(self, seqname):
+        import re
+        # Ensure the sequence name is valid
+        if not re.match(r'^[a-zA-Z_][a-zA-Z0-9_$]*$', seqname):
```

Comment:
It may be a good idea to compile the regex outside the function or use `utils.re_compile`.